src/ebay-fake-menu-button/menu-button.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { cloneElement, FC, useEffect, useRef, useState } from 'react'
+import classnames from 'classnames'
+import { filterByType, findComponent } from '../common/component-utils'
+import { randomId } from '../common/random-id'
+import { EbayButton, EbayButtonProps, EbayFakeMenu, EbayFakeMenuItemProps } from '..'
+import { EbayFakeMenuButtonItem, EbayFakeMenuButtonLabel, EbayFakeMenuButtonSeparator } from '.'
+
+export type MenuButtonProps = {
+    a11yText?: string;
+    className?: string;
+    onCollapse?: () => void;
+    onExpand?: () => void;
+    text?: string;
+}
+
+const EbayMenuButton: FC<EbayButtonProps & MenuButtonProps> = ({
+    a11yText,
+    borderless,
+    children,
+    className,
+    onCollapse = () => {},
+    onExpand = () => {},
+    text = '',
+    ...rest
+}) => {
+    const [expanded, setExpanded] = useState(false)
+    const [menuId, setMenuId] = useState<string|undefined>()
+    const firstItemRef = useRef<HTMLAnchorElement>()
+
+    const label = findComponent(children, EbayFakeMenuButtonLabel) || text
+    const menuItems = filterByType(children, [EbayFakeMenuButtonItem, EbayFakeMenuButtonSeparator])
+
+    useEffect(() => {
+        if (expanded === true) onExpand()
```

Comment:
@darkwebdev can we simplify with ternary
expanded ? onExpand() : onCollapse()
would that wont work?

Reply:
ok, I've fixed so that it makes more sense:
by default `expanded` is `undefined` so first time `useEffect` is called nothing should happen.